mica/stats/guide_stats.py

```diff
@@ -0,0 +1,552 @@
+import os
+import time
+import sys
+import argparse
+import numpy as np
+import tables
+from scipy.stats import scoreatpercentile
+from astropy.table import Table
+import logging
+import warnings
+
+from Chandra.Time import DateTime
+import agasc
+from kadi import events
+from Ska.engarchive import fetch, fetch_sci
+import mica.archive.obspar
+from mica.starcheck.starcheck import get_starcheck_catalog_at_date
+import Ska.quatutil
+import Ska.astro
+from mica.quaternion import Quat
+from chandra_aca import dark_model
+
+# Ignore known numexpr.necompiler and table.conditions warning
+warnings.filterwarnings(
+    'ignore',
+    message="using `oa_ndim == 0` when `op_axes` is NULL is deprecated.*",
+    category=DeprecationWarning)
+
+
+logger = logging.getLogger('star_stats')
+logger.setLevel(logging.INFO)
+if not len(logger.handlers):
+    logger.addHandler(logging.StreamHandler())
+
+STAT_VERSION = 0.1
+
+GUIDE_COLS = {
+    'obs': [
+        ('obsid', 'int'),
+        ('obi', 'int'),
+        ('guide_datestart', 'S21'),
+        ('guide_tstart', 'float'),
+        ('kalman_datestop', 'S21'),
+        ('revision', 'S15')],
+    'cat': [
+        ('slot', 'int'),
+        ('idx', 'int'),
+        ('type', 'S5'),
+        ('yang', 'float'),
+        ('zang', 'float'),
+        ('sz', 'S4'),
+        ('mag', 'float')],
+    'stat': [
+        ('n_samples', 'int'),
+        ('not_tracking', 'int'),
```

Comment:
Gotcha.  I'm already getting a little confused by `n_kalman` where you mean it as the number of samples that would be used in the Kalman filter and I'm trying not to think of it as the number of samples during the Kalman interval.